### ISSUE TITLE
Kotlin language server support

### ISSUE THREAD

**Issue by schmee** [ycm-core/ycmd#1137]
Would you consider a PR that adds support for Kotlin via https://github.com/fwcd/KotlinLanguageServer?

**Comment by bstaletic** [comment]
Sure, PRs are very welcome. If you need any help, feel free to ask on https://github.com/Valloric/ycmd. Also, don't forget the tests.

**Comment by Valloric** [comment]
@schmee Yes, Kotlin support would be very welcome!

(Also, @bstaletic was trying to recommend joining our dev Gitter room at https://gitter.im/Valloric/ycmd but he pasted the wrong link :D)

**Comment by puremourning** [comment]
FWIW From the readme:

> The original author created this project while he was considering using Kotlin in his work. He ended up deciding not to and is not really using Kotlin these days though this is a pretty fully-functional language server that just needs someone to use it every day for a while and iron out the last few pesky bugs.

I'm always wary of unsupported or partially supported upstream completers...

**Comment by puremourning** [comment]
We are currently working on [significantly reducing the overhead in adding new LSP based completers](https://github.com/puremourning/ycmd-1/commits/yaml).

When this is complete we will review which servers offer value and stability. We'll keep this in mind.

**Comment by bstaletic** [comment]
Kotlin server doesn't support jdk 11+ - https://github.com/fwcd/KotlinLanguageServer/pull/109

If someone wants to try the Kotlin server with YCM and has jdk older than 11, just add the following to vimrc:

```viml
let g:ycm_language_server={
    \ 'name': 'kotlin',
    \ 'filetypes': [ 'kotlin' ],
    \ 'cmdline': command_line,
    \ }
```

**Comment by bstaletic** [comment]
I managed to get kotlin server working with java8. They have just merged fwcd/KotlinLanguageServer#120 which makes the server work mostly fine (there are some curious KLS related bugs).

The full guide to setting up Kotlin is in ycm-core/lsp-examples#3, but essentially the configuration I mentioned above works.

What currently is not possible with `ycm_language_server` option is support for `FixIt`s. If there's enough demand for kotlin, we can promote the completer to an officially supported one and add `FixIt` support and tests.

pinging @apemost from ycm-core/YouCompleteMe#3350

**Comment by puremourning** [comment]
As kotlin is mostly working, let's close this, based on above. If there is strong support for features that aren't free, we can raise a new issue to track them. Thanks for the work @bstaletic
